kolibri/plugins/learn/assets/src/views/QuizRenderer/QuizReport.vue

```diff
@@ -0,0 +1,175 @@
+<template>
+
+  <ExamReport
+    :examAttempts="quizAttempts"
+    class="report"
+    :exam="{ title: 'test' }"
+    :userName="userName"
+    :userId="userId"
+    :currentInteractionHistory="currentInteractionHistory"
+    :currentInteraction="currentInteraction"
+    :selectedInteractionIndex="selectedInteractionIndex"
+    :questionNumber="questionNumber"
+    :exercise="{ available: available, kind: kind, files: files, extra_fields: extraFields }"
+    :itemId="itemId"
+    :completionTimestamp="completionTimestamp"
+    :complete="true"
+    :navigateToQuestion="navigateToQuestion"
+    :navigateToQuestionAttempt="navigateToQuestionAttempt"
+    :questions="questions"
+    :retry="true"
+    :timeSpent="timeSpent"
+    @repeat="$emit('repeat')"
+  />
+
+</template>
+
+
+<script>
+
+  import ExamReport from 'kolibri.coreVue.components.ExamReport';
+  import { MasteryLogResource } from 'kolibri.resources';
+
+  export default {
+    name: 'QuizReport',
+    components: {
+      ExamReport,
+    },
+    props: {
+      contentId: {
+        type: String,
+        required: true,
+      },
+      userId: {
+        type: String,
+        required: true,
+      },
+      userName: {
+        type: String,
+        required: true,
+      },
+      questions: {
+        type: Array,
+        default: () => [],
+      },
+      kind: {
+        type: String,
+        required: true,
+      },
+      files: {
+        type: Array,
+        default: () => [],
+      },
+      available: {
+        type: Boolean,
+        default: false,
+      },
+      extraFields: {
+        type: Object,
+        default: () => ({}),
+      },
+    },
+    data() {
+      return {
+        questionNumber: 0,
+        selectedInteractionIndex: 0,
+        pastTries: [],
+        currentTry: null,
+        masteryLogId: null,
+      };
+    },
+    computed: {
+      pastattempts() {
+        return this.currentTry ? this.currentTry.attemptlogs : [];
+      },
+      completionTimestamp() {
+        const time = this.pastattempts.length
+          ? this.pastattempts
+              .map(a => a.end_timestamp)
+              .sort()
+              .slice(-1)[0]
+          : null;
+        return time ? new Date(time) : null;
```

Comment:
Javascript data is notoriously bad for parsing dates, especially with cross-browser inconsistencies, but I think we may actually do this in other places too, so I have no recommendation at this time.

Reply:
Yeah - I'm not aware of any issues we've had with it, I think it's been somewhat mitigated by the fact that all of our timestamps from the backend are full ISO strings, and implementations are _mostly_ consistent for that.